code/modules/projectiles/guns/energy/special.dm

```diff
@@ -98,6 +98,13 @@
 	max_mod_capacity = 0
 	empty_state = null
 
+/obj/item/gun/energy/kinetic_accelerator/crossbow/update_icon()
+	if(!can_shoot())
+		add_overlay("[icon_state]_empty")
+	else
+		cut_overlays()

```

Comment:
Consider using `empty_state` instead of overhauling the proc so it can inherit.

Reply:
Its just awkward, ebows were made a child of KA's but they still use the traditional _empty icon setting.

Part of me thinks I should just kill empty_state since we only actually use it for the KA and it's just a weird exception.

Reply:
I would kill it and use this proc as the KA update_icon (which would inherit into ebows).

The KA uses _empty too, i'm not sure why they bothered to make it a var unless that was an attempt at optimization.

Reply:
It probably was but it definitely never caught on.